docs/architecture/adr-010-unified-cli-arguments-hermes.md

```diff
@@ -0,0 +1,182 @@
+* Flags which can be shorten and still be meaningful should be shorten. This is done for `connection`, `channel` and `sequence`, which become respectively `conn`, `chan` and `seq`.
+* In cases where there are undirectional parameters for a same object, the flags end with the suffix `-a` and `-b`.
+
+The following commands are implemented, with the binary name `hermes` omitted:
+
+### Hermes global flags
+
+* `hermes --config <CONFIG> <COMMAND>`
+
+* `hermes --json <COMMAND>`
+
+### Commands for clients
+
+* `create client --host-chain <HOST_CHAIN_ID> --reference-chain <REFERENCE_CHAIN_ID>`
+    * Optional: `[--clock-drift <CLOCK_DRIFT>] [--trust-threshold <TRUST_THRESHOLD>] [--trusting-period <TRUSTING_PERDIOD>]`
+
+* `update client --chain <CHAIN_ID> --client <CLIENT_ID>`
+    * Optional: `[--target-height <TARGET_HEIGHT>] [--trusted-height <TRUSTED_HEIGHT>]`
+
+* `upgrade client --chain <CHAIN_ID> --client <CLIENT_ID>`
+
+* `upgrade clients --chain <CHAIN_ID>`
+
+### Create a connection
+
+* `create connection --chain-a <CHAIN_A_ID> --chain-b <CHAIN_B_ID>`
+    * Optional: `[--delay <DELAY>]`
+
+* `create connection --chain-a <CHAIN_A_ID> --client-a <CLIENT_A_ID> --client-b <CLIENT_B_ID>`
+    * Optional: `[--delay <DELAY>]`
+
+### Create a channel
+
+* `create channel --chain-a <CHAIN_A_ID> --chain-b <CHAIN_B_ID> --port-a <PORT_A_ID> --port-b <PORT_B_ID>`
+    * Optional: `[--chan-version <VERSION>] [--new-client-conn] [--order <ORDER>]`
+
+* `create channel --chain-a <CHAIN_A_ID> --conn-a <CONNECTION_A_ID> --port-a <PORT_A_ID> --port-b <PORT_B_ID>`
+    * Optional: `[--chan-version <VERSION>] [--new-client-conn] [--order <ORDER>]`
+
+### Commands for keys
+
+* `keys add --chain <CHAIN_ID> --key-file <KEY_FILE> --mnemonic-file <MNEMONIC_FILE>`
+    * Optional: `[--hd-path <HD_PATH>] [--key-name <KEY_NAME>]`
+
+* `keys balance --chain <CHAIN_ID>`
+    * Optional: `[--key-name <KEY_NAME>]`
+
+* `keys delete --chain <CHAIN_ID> --all`
+
+* `keys delete --chain <CHAIN_ID> --key-name <KEY_NAME>`
+
+* `keys list --chain <CHAIN_ID>`
+
+### Listen
+
+* `listen --chain <CHAIN_ID>`
+    * Optional: `[--event <EVENT>]`
+
+### Misbehaviour
+
+* `misbehaviour --chain <CHAIN_ID> --client <CLIENT_ID>`
+
+### Start the relayer in multi-chain mode
+
+* `start`
+    * Optional: `[--full-scan]`
+
+### Clear objects
+
+* `clear packets --chain <CHAIN_ID> --port <PORT_ID> --chan <CHANNEL_ID>`
+
+### Queries
+
+__Channel__
+
+* `query channel client --chain <CHAIN_ID> --port <PORT_ID> --chan <CHANNEL_ID>`
+
+* `query channel end --chain <CHAIN_ID> --port <PORT_ID> --chan <CHANNEL_ID>`
+    * Optional: `[--height <HEIGHT>]`
+
+* `query channel ends --chain <CHAIN_ID> --port <PORT_ID> --chan <CHANNEL_ID>`
+    * Optional: `[--height <HEIGHT>] [--verbose]`
+
+* `query channels --host-chain <HOST_CHAIN_ID>`
+    * Optional: `[--reference-chain <REFERENCE_CHAIN_ID>] [--verbose]`
```

Comment:
`reference` doesn't work well here. I like `counterparty` in this case, something like:
```suggestion
* `query channels --chain <CHAIN_ID>`
    * Optional: `[--counterparty-chain <COUNTERPARTY_CHAIN_ID>] [--verbose]`
```